### ISSUE TITLE
Don't use current time in unit tests OR account for dates being bumped into next day due to UTC offsets, timezone transitions, etc

### ISSUE THREAD

**Issue by tn3rb** [eventespresso/event-espresso-core#656]
One of the JS unit tests in `assets/src/components/form/select/test/build-options.js` does not account for UTC offset correctly because running tests late in the day in Pacific time results in dates being generated for the next day in UTC. The test assumes that the generated datetimes start and end on the same day, but because the unit tests on Travis run in UTC, the end date for the test gets bumped to the next day after having the dates modified by adding time.

Here's an example of the failed test output:

```js
 ● buildOptions() › returns expected values for options using default optionsEntityMap for DatetimeSelect
expect(received).toEqual(expected)

Expected value to equal:
      [{"label": "DateTime 1 (2018-08-17 23:48:07 - 00:48:07)", "value": 30}]
Received:
      [{"label": "DateTime 1 (2018-08-17 23:48:07 - 2018-08-18 00:48:07)", "value": 30}]

at Object.<anonymous> (assets/src/components/form/select/test/build-options.js:100:72)
```

So either one of two things need to happen (and this type of logic needs to always be used in any similar future tests):

- tests continue to generate dates using the current time, but first detect if times pass a certain threshold and adjust the expected results when/if adding additional time duratios. ie: if date 1's time is on or after 11pm, then adding 1 or more hours to date 1 will result in date 2 being on the next day

or

- don't use the current time, and write tests using specific known dates and times so that the expected results can be determined correctly. ie: set the time to 3pm on date 1 so that adding 1 hour will always result in a time of 4pm, and not cause the date portion of date 2 to be bumped into the next day.

Because option 2 is less dynamic, it is less likely to uncover flaws in our code's logic that do not properly account for date changes as a result of UTC offsets, timezone transitions, etc. However, if those same things are not accounted for in the expected results, then the tests incorrectly fail under normal conditions. IMHO it is better to have well defined tests that do not throw false positives and then write new tests to account for edge cases that get reported by users and/or discovered during development.

**Comment by sethshoultes** [comment]
@tn3rb is this something that should be resolved right away or added to the backlog? It sounds like it needs to be fixed sooner rather than later. Also, are you asking me to make a decision on which option to go with? If so, it sounds like we should go with option one.

**Comment by tn3rb** [comment]
I was only passing to you to triage for Darren, since it is his code that is affected. I didn't want to assign to him directly without it going through you first or at least having you be aware of it. It should happen sooner rather than later but it's also not critical since it only affects unit tests. I can simply run tests again in the morning when this happens so it's really just a PITA as opposed to any kind of serious issue, but it does interrupt the flow of things.

**Comment by nerrad** [comment]
The particular test mentioned in this issue has already been fixed it looks like since it's now using a hardcoded time for the test.  I'm pretty sure I've been more diligent since this issue was created in making sure date tests are using static times.

I did a review of past builds on master and in pulls for EE core to see if there were any more recentish js test fails related to datetime issues and I didn't see any.  So I think we can close this issue.  If another such test fail pops up we can just create an issue/pull specific to it and get it fixed.
